src/sdam/topology.rs

```diff
@@ -522,8 +524,8 @@ impl TopologyWorker {
                     s.pool.mark_as_ready().await;
                 }
             }
-            self.publisher.publish_new_state(latest_state)
```

Comment:
we always publish a new state, regardless of whether there were material changes made. This is required so that servers blocked in server selection will continue to wake up every minHeartbeatFrequency and requesting checks instead of just blocking until they time out. A similar change was made in the 2.2.x branch, though that was done at the monitor level rather than topology.